### DISCUSSION TITLE
Using libuv with regional encoding in file and/or directory name

### DISCUSSION THREAD

**Discussion by RuProger** [libuv/libuv#3845] [Q&A]
Ok, guys. When i need to remove file or directory (for example, to other methods this actual too) with symbols in regional encoding, this is not works in Visual Studio 2015 with libuv v.1.9.1 (x86) (may be newer version is exists, but i can't found this anywhere).

This example works successful:

```
int main()
{
	char path[] = "c:/123/Тест.txt"; //sample string with regional encoding
	remove(path);
	return 0;
}
```

But when i try to use libuv, this decision is not works:

```
#include <uv.h>
#include <stdio.h>
#include <conio.h>
#pragma comment(lib, "libuv")

uv_loop_t* loop;
uv_fs_t request;

void remove_file_cb(uv_fs_t* req)
{
	if (!req->result)
	{
		printf("File removed successfully!\n");
	}
	else
	{
		printf("Error: %s\n", uv_strerror(req->result));
	}
	uv_fs_req_cleanup(&request);
	_getch();
}

int main()
{
	char path[] = "c:/123/Тест.txt"; //sample string with regional encoding
	loop = uv_default_loop();
	uv_fs_unlink(loop, &request, path, remove_file_cb);
	uv_run(loop, UV_RUN_DEFAULT);
	return 0;
}
```
Using setlocale and other encoding of source file - unfortunately not changes... Output is "Error: no such file or directory". Can libuv use filenames with regional encoding?

**Comment by bnoordhuis** [comment]
Make sure your source file/input string is UTF-8. Libuv takes care of converting it.

**Comment by RuProger** [comment]
Saving source file in Visual Studio in UTF-8 with signature is not helps, but without signature this works fine! Thanks! Now i'm looking for how to use strings in UTF-8 without saving source file in this encoding.
